activemq-stomp/src/test/java/org/apache/activemq/transport/stomp/Stomp11Test.java

```diff
@@ -1157,6 +1157,37 @@ public void testStompHeaderWithUndefinedEscapeSequence() throws Exception {
         assertTrue(stompFrame.getHeaders().get("message").equals("Undefined escape sequence [\\x] found in header!"));
     }
 
+    @Test(timeout = 60000)
+    public void testStompHeaderWithUndefinedEscapeSequenceInAllowedConnectHeader() throws Exception {
+        String connectFrame = "CONN\\xECT\n" +
```

Comment:
it's not clear to me why `\\x` is just ignored here

Reply:
Oh hmm.. I misread your interpretation of the spec as us not reading escaped characters in the CONNECT frame.

What do you think the previous patch was missing? The code I have touched deals with CONNECT/CONNECTED _decoding_ whereas the code you referenced deals with _encoding_ on the response.

Reply:
Hmmm. I referenced CONNECT frame's headers.

This is what I would expect to work without any attempt to decode the password:
```
CONNECT
login:ololo
passcode:bla\h\blah\na\\h
...
```

And this is something weird, that should not result in CONNECTED:)
```
CONN\\xECT
...
```

Reply:
Gotcha, I'm less confused now! I think I woke up too early today 😅